src/core/public/saved_objects/saved_objects_client.ts

```diff
@@ -45,7 +45,11 @@ import { HttpFetchOptions, HttpSetup } from '../http';
 
 type SavedObjectsFindOptions = Omit<
   SavedObjectFindOptionsServer,
-  'sortOrder' | 'rootSearchFields' | 'typeToNamespacesMap'
+  | 'sortOrder'
+  | 'rootSearchFields'
+  | 'typeToNamespacesMap'
+  | 'ACLSearchParams'
```

Comment:
There are two newly added: `ACLSearchParams` and `workspacesSearchOperator`. Can we make sure that if no feature flag is enabled, it doesn't break the current functionality?

Reply:
Sure. I've tried in my local cluster. The `ACLSearchParams` only be used in the new added `WorkspaceSavedObjectClientWrapper`, it won't change any existing logic if these options not provided. If feature flag is disabled, the saved objects without `workspaces` and `permissions` still can be find and return. 
The saved objects with `workspaces` or `permissions` property will using ACLSearch logic to find all saved objects. Here is the details:
1. ACLSearchParams provided
It will return all saved objects that match `ACLSearchParams.principals` and `ACLSearchParams.permissionModes`.

2. ACLSearchParams provided and workspacesSearchOperator === "OR" and options.workspaces provided
It will combine all saved objects that match `ACLSearchParams` and saved objects in the `options.workspaces`.